sdk/storage/src/datalake/path_client.cpp

```diff
@@ -0,0 +1,440 @@
+        std::move(result.Permissions),
+        std::move(result.ACL),
+        std::move(result.LeaseDuration),
+        std::move(result.LeaseState),
+        std::move(result.LeaseStatus),
+        ParseDataLakePropertiesString(result.Properties)};
+  }
+
+  PathLeaseResponse PathClient::Lease(const PathLeaseOptions& options) const
+  {
+    DataLakeRestClient::Path::LeaseOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.XMsLeaseAction = options.LeaseAction;
+    protocolLayerOptions.ProposedLeaseIdOptional = options.ProposedLeaseId;
+    protocolLayerOptions.XMsLeaseDuration = options.LeaseDuration;
+    protocolLayerOptions.XMsLeaseBreakPeriod = options.LeaseBreakPeriod;
+    protocolLayerOptions.LeaseIdOptional = options.LeaseId;
+    protocolLayerOptions.IfMatch = options.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
+    protocolLayerOptions.Timeout = options.Timeout;
+    return DataLakeRestClient::Path::Lease(
+        m_pathUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  ReadPathResponse PathClient::Read(const PathReadOptions& options) const
+  {
+    DataLakeRestClient::Path::ReadOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Range = options.Range;
+    protocolLayerOptions.XMsRangeGetContentMd5 = options.RangeGetContentMd5;
+    protocolLayerOptions.LeaseIdOptional = options.LeaseId;
+    protocolLayerOptions.IfMatch = options.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
+    protocolLayerOptions.Timeout = options.Timeout;
+    auto result = DataLakeRestClient::Path::Read(
+        m_pathUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+    return ReadPathResponse{
+        std::move(result.BodyBuffer),
+        std::move(result.AcceptRanges),
+        std::move(result.CacheControl),
+        std::move(result.ContentDisposition),
+        std::move(result.ContentEncoding),
+        std::move(result.ContentLanguage),
+        std::move(result.ContentLength),
+        std::move(result.ContentRange),
+        std::move(result.ContentType),
+        std::move(result.ContentMD5),
+        std::move(result.Date),
+        std::move(result.ETag),
+        std::move(result.LastModified),
+        std::move(result.RequestId),
+        std::move(result.Version),
+        std::move(result.ResourceType),
+        std::move(result.LeaseDuration),
+        std::move(result.LeaseState),
+        std::move(result.LeaseStatus),
+        std::move(result.XMsContentMd5),
+        ParseDataLakePropertiesString(result.Properties)};
```

Comment:
I feel this is really hard to read. I have to go to the definition of `ReadPathResponse` and count from top to know which member we're assigning value to.

Reply:
But this avoids tens of constructor calls for `std::string`. It's C++ and we don't have named argument so we have to live with it.

Reply:
What the avoidance of tens of construct calls for `std::string` can bring to us?

Reply:
I don't think this conversation is resolved.

Reply:
Added a topic for discussion. Please resolve it if you think it can be overlooked for now and revisit when we have a decision.